### DISCUSSION TITLE
Version conflict between Zigbee2MQTT and node JS

### DISCUSSION THREAD

**Discussion by Herwigvd** [Koenkk/zigbee2mqtt#13364] [General]
I installed the required software on a fresh RPI4 (SD-card OS imager_1.5.exe)
Afterwards, I installed mosquitto and zigbee2mqtt
I changed the required configuration files (mosquitto.conf and configuration.yaml).
I have a working configuration now.
I can read an Xiaomi Aquara sensor and can control an philips hue bulb.

But when I start zigbee2mqtt with 'sudo npm start', I get the message:
**Zigbee2MQTT requires node version ^14 || ^16 || ^17 || ^18, you are running V12.22.12**

In several ways, I tried to update nodejs but I always get stuck.
Can someone please advise?

This is the list of installation commands I used 
---------------------------------------------------
$ sudo apt install -y mosquitto mosquitto-clients
$ sudo systemctl enable mosquitto.service

$ sudo apt-get install -y nodejs npm git make g++ gcc
$ git clone https://github.com/Koenkk/zigbee2mqtt.git
$ sudo mv zigbee2mqtt /opt/zigbee2mqtt
$ cd /opt/zigbee2mqtt
$ npm ci
$ sudo npm start

**Comment by lux73** [comment]
you should use node.js repository and not default -> https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-debian-10

don't know which Distro you exactly use but the Steps are similar

**Reply by Herwigvd** [reply]
Thanks for your feedback.
Version is Rasbian GNU/Linux 11

I installed nodejs as you recommended (Using a PPA software repository) but encounter an error when I execute 'sudo npm start'
I always run into this error when I try to install a different version of nodejs.
Up to now, only the default version works, but with remarks about not up to date version.

Error has to do with libnode.so.72 that can not be opened or found.
I included the log of the installation of nodejs and startup with error message.

Any suggestion how to solve this error?

![Startup error](https://user-images.githubusercontent.com/108485192/184215335-e5e7e815-2012-4ed4-8db2-8804634ddcdf.png)

**Reply by lux73** [reply]
`apt install libnode72` should fix this

**Reply by lux73** [reply]
make sure you have deinstalled old bundled nodejs Version so your System hast only new ppa Repo installed

**Reply by Herwigvd** [reply]
Ok. Thanks.
That worked. 
I had to restart from a fresh installed Pi to get rid of the old software.
